client/client_test.go

```diff
@@ -540,6 +540,166 @@ func TestHandleModifyResponse(t *testing.T) {
 			},
 			ClientError: "received a session parameter result when there was none pending",
 		}},
+	}, {
+		desc: "AckType set to FIB_ACK, receive AFTResult_RIB_PROGRAMMED and AFTResult_FIB_PROGRAMMED ",
+		inClient: &Client{
+			qs: &clientQs{
+				pendq: &pendingQueue{
+					Ops: map[uint64]*PendingOp{
+						1: {
+							Timestamp: 42,
+							Op:        &spb.AFTOperation{Id: 1},
+						},
+					},
+				},
+				sending: &atomic.Bool{},
+			},
+			state: &clientState{
+				SessParams: &spb.SessionParameters{
+					AckType: spb.SessionParameters_RIB_AND_FIB_ACK,
+				},
+			},
+		},
+		inResponse: &spb.ModifyResponse{
+			Result: []*spb.AFTResult{
+				{
+					Id:     1,
+					Status: spb.AFTResult_RIB_PROGRAMMED,
+				}, {
+					Id:     1,
+					Status: spb.AFTResult_FIB_PROGRAMMED,
+				}},
+		},
+		wantResults: []*OpResult{{
+			Timestamp:         42,
+			OperationID:       1,
+			ProgrammingResult: spb.AFTResult_RIB_PROGRAMMED,
+			Details:           &OpDetailsResults{},
+		}, {
+			Timestamp:         42,
+			OperationID:       1,
+			ProgrammingResult: spb.AFTResult_FIB_PROGRAMMED,
+			Details:           &OpDetailsResults{},
+		}},
+	}, {
+		desc: "AckType set to FIB_ACK, receive 1. AFTResult_FAILED 2. AFTResult_RIB_PROGRAMMED and AFTResult_FAILED ",
```

Comment:
Business logic wise, `AFTResult_FIB_FAILED` probably make more sense.

Reply:
EOS doesn't respond with AFTResult_FIB_FAILED, but according to Arista team, they should have some support for FIB_FAILED in a couple of weeks or so.

Reply:
But this is about client unittest behavior. It is vendor agnostic, right? :)

Reply:
Given what we defined at gribi.pb.go
https://source.corp.google.com/piper///depot/google3/blaze-out/genfiles/third_party/openconfig/gribi/v1/proto/service/gribi.pb.go;rcl=HEAD;l=179
looks like FIB_FAILED situation does not exist

Reply:
yes, will change